src/main.cpp

```diff
@@ -192,7 +192,14 @@ void main(void)
 	uint32_t *reset_vector;
 	reset_vector = (uint32_t *)RESET_VECTOR;
 	start_arc(*reset_vector);
```

Comment:
For this firmware application it should work if the ARC core is started after the tasks.
Though depending on the changes users make on the quark side it can make a difference because of the way the ARC core-libs initially setups some of the GPIO/peripherals.